tests/route/test_route_flap.py

```diff
@@ -137,7 +139,21 @@ def test_route_flap(duthost, tbinfo, ptfhost, ptfadapter,
     ptf_ip = tbinfo['ptf_ip']
     common_config = tbinfo['topo']['properties']['configuration_properties'].get('common', {})
     nexthop = common_config.get('nhipv4', NHIPV4)
+
+    # On dual-tor, unicast upstream l3 packet destination mac should be vlan mac
+    # After routing, output packet source mac will be replaced with port-channel mac (same as dut_mac)
+    # On dual-tor, vlan mac is different with dut_mac. U0/L0 use same vlan mac for AR response
+    # On single tor, vlan mac (if exists) is same as dut_mac
     dut_mac = duthost.facts['router_mac']
+    vlan_mac = dut_mac
```

Comment:
@jcaiMR 

Could we use try/except in this case:
try:
    vlan_mac = tbinfo['topo']['properties']['topology']['DUT']['vlan_configs']['default_vlan_config']['mac']
except KeyError:
    vlan_mac = dut_mac

Reply:
Use try catch is not strong enough, as some time, configuration may has default_vlan_config. And in default vlan config, it has multiple vlan interfaces, some vlan configuration may has empty mac address info, like 'mac':''. In this case we still need walk through all vlan interfaces and try to get the one who has vlan mac specified. 

But we can add try/catch for the protection of "tbinfo['topo']['properties']['topology']['DUT']['vlan_configs']" .
How about your thinking ?

Reply:
@jcaiMR  as we talked, may add a check for dual tor, which should use this Vlan mac, if it doesn't exist, drop an exception.

Reply:
Updated in the latest diff